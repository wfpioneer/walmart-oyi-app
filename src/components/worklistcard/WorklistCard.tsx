import React, {useState, useEffect} from 'react';
import {Text, View, StyleSheet} from 'react-native';
import styles from './WorklistCard.style';

/**
 * Basic worklist card component to be populated with worklist items as needed.
 */
export interface Props {
    goalTitle: string,
    goal: number,
    complete: number,
    completionPercentage: number,
    completionGoal?: number
};

function WorklistCard(Props:Props) {
    let target;
    Props.completionGoal?(target=Props.completionGoal):(target=95);
    const [atGoal] = useState(((Props.completionPercentage>=target)?styles.barFillAtGoal:styles.barFillNotAtGoal));
    return (
         <View style={styles.card}>
             <View style={styles.head}>
                 <Text style={styles.title}>{Props.goalTitle}</Text>
                 <Text style={styles.progress}>Goal {target}%</Text>
             </View>
             <View style={styles.progressBar}>
                 <View style={[StyleSheet.absoluteFill,atGoal, {width:`${Props.completionPercentage}%`}]} />
             </View>
             <Text style={styles.counter}>{Props.complete} of {Props.goal} items</Text>
         </View>
     )
 };

 export default WorklistCard;