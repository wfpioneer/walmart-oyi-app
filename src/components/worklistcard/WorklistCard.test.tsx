import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import WorklistCard from './WorklistCard';

jest.mock('react-native-vector-icons/MaterialCommunityIcons', () => 'mockMaterialCommunityIcons');

describe('testing OHQtyUpdate component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  describe('render component and test buttons', () => {
    const mockHandleSubmit = jest.fn();
    it('render component with some test data and goal not as zero', () => {
      const { toJSON } = render(
        <WorklistCard
          goalTitle="Test"
          goal={100}
          complete={90}
          completionPercentage={90}
          completionGoal={100}
          onPress={mockHandleSubmit}
          inProgress={false}
          pendingPercentage={0}
        />
      );
      expect(toJSON()).toMatchSnapshot();
    });
    it('render component with some test data and goal as zero', () => {
      const { toJSON } = render(
        <WorklistCard
          goalTitle="Test"
          goal={0}
          complete={90}
          completionPercentage={90}
          completionGoal={100}
          onPress={mockHandleSubmit}
          inProgress={false}
          pendingPercentage={0}
        />
      );
      expect(toJSON()).toMatchSnapshot();
    });

    it('render component with inProgress Bar enabled', () => {
      const { toJSON } = render(
        <WorklistCard
          goalTitle="Test"
          goal={100}
          complete={75}
          completionPercentage={75}
          completionGoal={100}
          onPress={mockHandleSubmit}
          inProgress={false}
          pendingPercentage={25}
        />
      );
      expect(toJSON()).toMatchSnapshot();
    });

    it('should call mock func on card click', () => {
      const { getByTestId } = render(
        <WorklistCard
          goalTitle="Test"
          goal={70}
          complete={90}
          completionPercentage={90}
          completionGoal={100}
          onPress={mockHandleSubmit}
          inProgress={false}
          pendingPercentage={0}
        />
      );
      const btnCard = getByTestId('btnCard');
      fireEvent.press(btnCard);

      expect(mockHandleSubmit).toHaveBeenCalled();
    });
  });
});
