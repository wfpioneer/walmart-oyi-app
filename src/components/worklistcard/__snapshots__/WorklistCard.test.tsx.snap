// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing OHQtyUpdate component render component and test buttons render component with inProgress Bar enabled 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#ffffffff",
      "borderRadius": 10,
      "elevation": 2,
      "margin": 4,
      "opacity": 1,
      "padding": 6,
      "width": "95%",
    }
  }
  testID="btnCard"
>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        {
          "flex": 1,
        }
      }
    >
      Test
    </Text>
    <Text
      style={
        {
          "flex": 1,
          "fontWeight": "bold",
          "textAlign": "right",
        }
      }
    >
      [missing "en.GENERICS.GOAL" translation] 100%
    </Text>
  </View>
  <View
    style={
      {
        "backgroundColor": "#e0e0e0",
        "borderRadius": 10,
        "flexDirection": "row",
        "height": 10,
        "marginBottom": 5,
        "marginTop": 5,
        "width": "100%",
      }
    }
  >
    <View
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "backgroundColor": "#D6512D",
            "borderRadius": 10,
          },
          {
            "width": "75%",
          },
        ]
      }
    />
  </View>
  <Text
    style={
      {
        "color": "#006DE5",
        "textAlign": "right",
      }
    }
  >
    [missing "en.HOME.WORKLIST_GOAL_COMPLETE" translation]
  </Text>
</View>
`;

exports[`testing OHQtyUpdate component render component and test buttons render component with some test data and goal as zero 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#ffffffff",
      "borderRadius": 10,
      "elevation": 2,
      "margin": 4,
      "opacity": 1,
      "padding": 6,
      "width": "95%",
    }
  }
  testID="btnCard"
>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        {
          "flex": 1,
        }
      }
    >
      Test
    </Text>
    <Text
      style={
        {
          "flex": 1,
          "fontWeight": "bold",
          "textAlign": "right",
        }
      }
    >
      [missing "en.GENERICS.GOAL" translation] 100%
    </Text>
  </View>
  <View
    style={
      {
        "backgroundColor": "#e0e0e0",
        "borderRadius": 10,
        "flexDirection": "row",
        "height": 10,
        "marginBottom": 5,
        "marginTop": 5,
        "width": "100%",
      }
    }
  >
    <View
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "backgroundColor": "#e0e0e0",
            "borderRadius": 10,
          },
          {
            "width": "90%",
          },
        ]
      }
    />
  </View>
  <Text
    style={
      {
        "color": "#006DE5",
        "textAlign": "right",
      }
    }
  >
    [missing "en.HOME.WORKLIST_GOAL_COMPLETE" translation]
  </Text>
</View>
`;

exports[`testing OHQtyUpdate component render component and test buttons render component with some test data and goal not as zero 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#ffffffff",
      "borderRadius": 10,
      "elevation": 2,
      "margin": 4,
      "opacity": 1,
      "padding": 6,
      "width": "95%",
    }
  }
  testID="btnCard"
>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        {
          "flex": 1,
        }
      }
    >
      Test
    </Text>
    <Text
      style={
        {
          "flex": 1,
          "fontWeight": "bold",
          "textAlign": "right",
        }
      }
    >
      [missing "en.GENERICS.GOAL" translation] 100%
    </Text>
  </View>
  <View
    style={
      {
        "backgroundColor": "#e0e0e0",
        "borderRadius": 10,
        "flexDirection": "row",
        "height": 10,
        "marginBottom": 5,
        "marginTop": 5,
        "width": "100%",
      }
    }
  >
    <View
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "backgroundColor": "#D6512D",
            "borderRadius": 10,
          },
          {
            "width": "90%",
          },
        ]
      }
    />
  </View>
  <Text
    style={
      {
        "color": "#006DE5",
        "textAlign": "right",
      }
    }
  >
    [missing "en.HOME.WORKLIST_GOAL_COMPLETE" translation]
  </Text>
</View>
`;
